### PR TITLE
WIP: remove use of inspect.getargspec for Python 3.x (deprecated in 3.5)

### PR DESCRIPTION
**Pull request by rgommers**
Not quite finished yet, but already filing a PR to ensure not forgetting this for 0.16.x. That's the only thing that seems to need fixing with Python 3.5.0b2

Didn't touch `doc/source/scipyoptdoc.py` and `scipy/_lib/decorator.py` yet, did file an issue on the `decorator` issue tracker.

### CODE REVIEW

**Comment by rgommers** [comment]
Yet another Python 3.x "improvement" that generates work but doesn't really look much better. They've even deprecated ``getfullargspec`` again after introducing it in 3.x

**Comment by charris** [comment]
>Yet another Python 3.x "improvement" that generates work but doesn't really look much better.

The Python developers should be more like the Texas Legislature and meet for 140 days on odd numbered years ;)

File: scipy/_lib/_util.py [hunk 1]
```diff
@@ -140,3 +141,36 @@ def _asarray_validated(a, check_finite=True,
...
+    """
+    if sys.version_info[0] >= 3:
+        sig = inspect.signature(func)
+        args = sig.parameters
+        tmp_defaults = [d.default for d in sig.parameters.values()]
+        defaults = [d for d in tmp_defaults if d is not inspect._empty]
+    else:
+        args, _, _, defaults = inspect.getargspec(func)
+
+    return args, defaults
```
Comment (larsmans) [inline scipy/_lib/_util.py]: It should be marginally faster to conditionally define the function rather than check for the Python version in every call.

Comment (rgommers) [inline scipy/_lib/_util.py]: agree, will change

**Comment by rgommers** [comment]
This is still WIP and I have no time to finish it up before my holiday, so let's hold off on this one. Would have been nice, but not a blocker.

**Comment by ev-br** [comment]
An alternative version: gh-5248

**Comment by rgommers** [comment]
closing this in favor of gh-5248, which is the better approach.

**Pull request closed without merging**